src/Albany_Application.cpp

```diff
@@ -2498,51 +2498,110 @@ Application::evaluateResponseDistParamHessian_pp(
     const std::string&                      param_name,
     const Teuchos::RCP<Thyra_LinearOp>&     H)
 {
-  Teuchos::ParameterList coloring_params;
-  std::string matrixType = "Hessian";
-  coloring_params.set("matrixType", matrixType);
-  coloring_params.set("symmetric", true);
+  int l1;
+  bool l1_is_distributed;
+  Albany::getParameterVectorID(l1, l1_is_distributed, param_name);
 
   // Get the crs Hessian:
   RCP<Tpetra_CrsMatrix> Ht = Albany::getTpetraMatrix(H);
   Ht->resumeFill();
 
-  // Create a colorer
-  Zoltan2::TpetraCrsColorer<Tpetra_CrsMatrix> colorer(Ht);
+  bool replace_by_I = 
+    problemParams->sublist("Hessian").sublist(Albany::strint("Response", response_index)).isParameter(Albany::strint("Replace H_pp by I parameter", parameter_index)) ?
```

Comment:
I would make this more verbose:
"Replace H_pp with Identity for Parameter"

Reply:
It is better to validate the "Hessian" and "Response X" sublists to avoid typos.

Reply:
> I would make this more verbose: "Replace H_pp with Identity for Parameter"

Or even the shorter "Replace H_pp with Identity" (the part "for Parameter" is kind of implied by the "_pp" part).

Sorry, I'm a fan of short parameter list keys. Feel free to ignore this comment. :)

Reply:
Ah, I misread the code. I didn't notice it was "strint-ed" with the parameter index. Then I agree with you.